ui/components/ui/unit-input/unit-input.component.js

```diff
@@ -179,7 +179,16 @@ export default class UnitInput extends PureComponent {
               onBlur={this.handleInputBlur}
               onFocus={this.handleInputFocus}
               onKeyDown={(e) => {
-                if (['e', 'E', '+', '-'].includes(e.key)) {
+                const isNumericInput = DECIMAL_INPUT_REGEX.test(e.key)
+                const isControlKey =
+                  e.key === 'Backspace' ||
+                  e.key === 'Tab' ||
+                  e.key === 'ArrowLeft' ||
+                  e.key === 'ArrowRight' ||
+                  e.key === 'Delete'
+
+                // Allow numeric input and decimal point
+                if (!isNumericInput && !isControlKey) {
                   e.preventDefault();
                 }
               }}
```

Comment:
Can we move this to a handler similar to what we do with the other attributes (e.g., `onBlur`, `onChange`)?